src/main/webapp/app/exam/participate/exercises/programming/code-editor/exam-code-editor-student-container.component.ts

```diff
@@ -1,124 +0,0 @@
-import { Component, Input, OnInit, ViewChild } from '@angular/core';
-import * as moment from 'moment';
-import { TranslateService } from '@ngx-translate/core';
-import { AlertService } from 'app/core/alert/alert.service';
-import { CodeEditorBuildOutputComponent } from 'app/exercises/programming/shared/code-editor/build-output/code-editor-build-output.component';
-import { ButtonSize } from 'app/shared/components/button.component';
-import { CodeEditorSessionService } from 'app/exercises/programming/shared/code-editor/service/code-editor-session.service';
-import { CodeEditorFileService } from 'app/exercises/programming/shared/code-editor/service/code-editor-file.service';
-import { CodeEditorActionsComponent } from 'app/exercises/programming/shared/code-editor/actions/code-editor-actions.component';
-import { CodeEditorAceComponent } from 'app/exercises/programming/shared/code-editor/ace/code-editor-ace.component';
-import { ExerciseType } from 'app/entities/exercise.model';
-import { StudentParticipation } from 'app/entities/participation/student-participation.model';
-import { CodeEditorContainerComponent } from 'app/exercises/programming/shared/code-editor/code-editor-mode-container.component';
-import { CodeEditorInstructionsComponent } from 'app/exercises/programming/shared/code-editor/instructions/code-editor-instructions.component';
-import { CodeEditorFileBrowserComponent } from 'app/exercises/programming/shared/code-editor/file-browser/code-editor-file-browser.component';
-import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
-import { CommitState, DomainType, FileChange } from 'app/exercises/programming/shared/code-editor/model/code-editor.model';
-import { DomainService } from 'app/exercises/programming/shared/code-editor/service/code-editor-domain.service';
-import { CodeEditorConflictStateService } from 'app/exercises/programming/shared/code-editor/service/code-editor-conflict-state.service';
-import { CodeEditorSubmissionService } from 'app/exercises/programming/shared/code-editor/service/code-editor-submission.service';
-import {
-    CodeEditorBuildLogService,
-    CodeEditorRepositoryFileService,
-    CodeEditorRepositoryService,
-} from 'app/exercises/programming/shared/code-editor/service/code-editor-repository.service';
-
-@Component({
-    selector: 'jhi-exam-code-editor-student',
-    templateUrl: './exam-code-editor-student-container.component.html',
-    providers: [
```

Comment:
I'm not sure if this would work with the projected components outside the code-editor-container, like result and instructions, since they would then have other service instances. As it works right now, I'm closing this.

Reply:
I agree to @kloessst that we should have unique instances of the services.
It took us a really long time and lots of debugging and testing to get this working correctly. 

Why do you change this now?

Reply:
I could add the services to the programming-exam-submission component as an extra, see my comment below. But as of functionality, this doesn't seem to be needed.

Reply:
We don't need it as we call ngOnInit now to set the right domain for the services when switching exercises. This was our first approach to solving our issues with mutliple programming exercises and it worked as far as I can tell.